lib/dao.js

```diff
@@ -1526,6 +1526,27 @@ function NumberType(val) {
   return !isNaN(num) ? num : val;
 }
 
+function coerceArray(val) {
+  if (Array.isArray(val)) {
+    return val;
+  }
+
+  if (!utils.isPlainObject(val)) {
+    throw new Error(g.f('Value is not an {{array}} or {{array-like object}}'));
+  }
+
+  var arrayVal = new Array(Object.keys(val).length);
```

Comment:
If I understand this correctly, then this is opening a security vulnerability, where the attacker can send an array-like object `{ 4000000000: true }` and we will create a 4GB array containing a single item at a large index. This is reverting the prevention measures implemented by qs.

https://github.com/strongloop/loopback/issues/2824#issuecomment-254535482
> Also in that Issue they reveal that arrayLength exists to prevent params like `key[999999]` auto-vivifying an array of that length, which I'll keep in mind to avoid as well.

I think we need to find a different solution here.